src/wallet/rpcwallet.cpp

```diff
@@ -2660,6 +2664,104 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
+            "\nFor the given account number, return a UA in accordance with the remaining arguments."
+            "\nIf no diversifier index is given (or the string \"*\"), the next unused index will be selected."
+            "\nCompute a unified address corresponding to the diversifier index, containing addresses"
+            "\nfor the given protocols (default all).\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"account\": n,                (numeric) next available account number\n"
+            "  \"diversifier\": n,            (numeric) the index specified or chosen\n"
+            "  \"pools\": [\"protocol\",...]\",  (json array of string) the protocols (e.g. \"transparent\", \"orchard\") the UA contains\n"
+            "  \"unifiedaddress\"             (string) The corresponding address\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("z_getaddressforaccount", "4")
+            + HelpExampleCli("z_getaddressforaccount", "4 1")
+            + HelpExampleCli("z_getaddressforaccount", "4 1 '[\"transparent\",\"sapling\",\"orchard\"]'")
+            + HelpExampleRpc("z_getaddressforaccount", "4")
+        );
+
+    if (!fExperimentalOrchardWallet) {
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: the Orchard wallet experimental extensions are disabled.");
+    }
+    int64_t account = params[0].get_int64();
+    if (account < 0 || account > 0x7fffffff-1) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid account number, must be 1 <= account <= (2^31)-2.");
+    }
+
+    //diversifier_t di;
+    std::vector<unsigned char> parsed_diversifier;
+    if (params.size() >= 2) {
+        if (params[1].getType() != UniValue::VNUM) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid diversifier index, must be an unsigned integer.");
+        }
+        ParseArbitraryInt(params[1].getValStr(), parsed_diversifier);
+        if (parsed_diversifier.size() > sizeof(diversifier_t)) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "diversifier index is too large.");
+        }
+    } else {
+        // TODO get next unused diversifier index from wallet
+    }
+    diversifier_t diversifier{};
+    // expand vector just to print, there is probably a better way
+    for (int i = 0; i < parsed_diversifier.size(); ++i) diversifier[i] = parsed_diversifier[i];
+    // diversifier_index_t di{diversifier};
+
+    UniValue protocols(UniValue::VARR);
+    if (params.size() >= 3) {
+        protocols = params[2].get_array();
+    } else {
+        // default is all
+        protocols.push_back("transparent");
+        protocols.push_back("sapling");
+        protocols.push_back("orchard");
+    }
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("account", account);
+    result.pushKV("diversifier", params[1].write());
+    // Should the JSON key be "protocols" instead of "pools"?
+    result.pushKV("pools", protocols);
```

Comment:
```suggestion
    result.pushKV("protocols", protocols);
```
This is called a "transfer protocol" in https://zips.z.cash/zip-0316#terminology